### PR TITLE
Change to Atomic Counters

### PR DESCRIPTION
**Pull request by ezhdanovskiy**
The primary mechanism for managing state in Go is
communication over channels. We saw this for example
with [worker pools](worker-pools). There are a few other
options for managing state though. Here we'll
look at using the `sync/atomic` package for _atomic
counters_ accessed by multiple goroutines.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,6 @@
 
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
+
+.idea
+tmp
```
Comment (unicikl) [inline .gitignore]: add new line

File: README.md [hunk 2]
```diff
@@ -1 +1,8 @@
-# prtests
\ No newline at end of file
+# prtests
+
+The primary mechanism for managing state in Go is
```
Comment (unicikl) [inline README.md]: Reformat it please

Comment (ezhdanovskiy) [reply README.md]: ok

File: main.go [hunk 3]
```diff
@@ -1,31 +1,49 @@
...
+				atomic.AddUint64(&ops, 1)
+
+				// Wait a bit between increments.
+				time.Sleep(time.Millisecond)
+			}
+		}()
+	}
+
+	// Wait a second to allow some ops to accumulate.
+	time.Sleep(time.Second)
```
Comment (unicikl) [inline main.go]: double duration please
```suggestion
	time.Sleep(time.Second*2)
```

File: main.go [hunk 4]
```diff
@@ -1,31 +1,49 @@
...
+				// `&` syntax.
+				atomic.AddUint64(&ops, 1)
+
+				// Wait a bit between increments.
+				time.Sleep(time.Millisecond)
+			}
+		}()
+	}
+
+	// Wait a second to allow some ops to accumulate.
```
Comment (unicikl) [inline main.go]: fix comment

**Review by unicikl — CHANGES_REQUESTED** [review]

**Review by unicikl — APPROVED** [review]
ok

**Pull request merged**